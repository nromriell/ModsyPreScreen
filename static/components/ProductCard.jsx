import React from 'react'

import '../style/theme.css'

/**
 * @author
 * Nathan Romriell - 10/16/2019
 *
 * @info
 * Functional Component - ProductCard
 *
 * @description
 * Render only component for displaying a product row
 *
 * @props
 * url                    - Link to product page
 * image                  - Object containing image information requires "href" attribute
 * name                   - Product name
 * price                  - Product price
 * description            - Product description
 *
 */
const ProductCard = (props) => {
    return (
    <a className="Card" href={props.url}>
        <div style={{backgroundImage:"url("+((props.image !== undefined && props.image !== null) ? props.image.href : "")+")"}} className="CardImage"/>
        <div className="CardTitle">{props.name}</div>
        <div className="CardPrice">{props.price}</div>
        <div className="CardDescription">{props.description}</div>
    </a>
    );
};

export default ProductCard;