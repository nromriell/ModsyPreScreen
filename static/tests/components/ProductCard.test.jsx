import React from 'react'
import renderer from 'react-test-renderer'
import ProductCard from '../../components/ProductCard'

describe('OnAppInit Tests', () => {
    test('Renders Correctly', () => {
       const component = renderer.create(<ProductCard/>);
       expect(component.toJSON()).toMatchSnapshot();
    });

});