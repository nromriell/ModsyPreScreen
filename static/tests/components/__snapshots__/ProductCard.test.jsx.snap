// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnAppInit Tests Renders Correctly 1`] = `
<a
  className="Card"
>
  <div
    className="CardImage"
    style={
      Object {
        "backgroundImage": "url()",
      }
    }
  />
  <div
    className="CardTitle"
  />
  <div
    className="CardPrice"
  />
  <div
    className="CardDescription"
  />
</a>
`;
